dape.el

```diff
@@ -581,61 +552,64 @@ The hook is run with one argument, the compilation buffer."
 
 (defmacro dape--callback (&rest body)
   "Create callback lambda for `dape-request' with BODY."
-  `(lambda (&optional process body success msg)
-     (ignore process body success msg)
+  `(lambda (&optional conn body success-p msg)
+     (ignore conn body success-p msg)
      ,@body))
 
 (defmacro dape--with (request-fn args &rest body)
```

Comment:
`dape--with` and `dape--callback` are odd unidiomatic macros.  `dape--callback` in particular is "anaphoric" meaning it establishes variables "magically".  I don't absolutely hate anaphoric macros but I don't like them either (Paul Graham's https://letoverlambda.com/ chapther 6 introduces some).  But I think it's better to keep them small and not problem-domain-specific, like here.

My  understanding you try to use these macro tricks to hide the ugliness from the `jsonrpc-async-request` CB nature (which your previous parser also needed).  But it doesn't really work in my opinion.  I can't reorder code naturally like I would in an imperative approach.

Nor to mention it's very tough to understand the flow if one of these requests fails.